### DISCUSSION TITLE
When I send a DM to the corde bot i get "Cannot send messages to this user" error

### DISCUSSION THREAD

**Discussion by habdelra** [cordejs/corde#1071] [Q&A] [answered]
I want to write a test for a bot that has the ability to send DM's. However, when I send a DM to the corde bot I get an error:
```
events:guild-message failed to run command 'dm-ping' with args:      at toReturn async Object.run (/Users/hassan/codez/cardstack/packages/cardpay-bot/commands/guild/dm-ping.js:12:14)
  events:guild-message     at async run (/Users/hassan/codez/cardstack/packages/cardpay-bot/events/guild-message.js:53:9) {
  events:guild-message   name: 'DiscordAPIError',
  events:guild-message   message: 'Cannot send messages to this user',
  events:guild-message   method: 'post',
  events:guild-message   path: '/users/896093538297708564/channels',
  events:guild-message   code: 50007,
  events:guild-message   httpStatus: 400
  events:guild-message }
```

specifically this command throws:
```js
let dm = await message.member.createDM();
```
This looks like the error you get when a user has their privacy settings set to not allow DM's from server members. When I interact with bot bot using a normal user it can send DM's just fine. Is there a setting that I need to specify to allow the corde bot to receive DM's?

**Comment by habdelra** [comment, marked as answer]
It looks like discord might not actually allow a bot to send a DM to another bot https://stackoverflow.com/questions/49266283/is-it-possible-to-have-one-discord-bot-dm-another-bot/49395167#:~:text=Both%20bots%20can%20receive%20DMs%20from%20users%20just%20fine.

**Reply by habdelra** [reply]
My approach is to basically do all the DM testing in a public channel with the `process.env.NODE_ENV` is `corde_test` using a helper function to return either a `DMChannel` or a `TextChannel` based on the env var that I use instead of `Message.member.createDM()`:
```ts
export async function createDM(message: Message): Promise<Message['channel']> {
  if (process.env.NODE_ENV === 'corde_test') {
    return message.channel;
  }
  if (!message.member) {
    throw new Error('Message must have a member');
  }
  return await message.member.createDM();
}
```

**Reply by lucasgmagalhaes** [reply]
Great! I just realise that corde's site doesn't seem to have a doc for that.
